lib/actions/write.js

```diff
@@ -67,30 +66,30 @@ function processAction(msg, cfg) {
             });
             const fileName = messageToEmit.id + '.csv';
             messageToEmit.attachments[fileName] = {
-                "content-type": "text/csv",
-                url: signedUrl.get_url
+                'content-type': 'text/csv',
+                'url': signedUrl.get_url
             };
             signedUrl = null;
             rowCount = 0;
 
             console.log('Emitting message %j', messageToEmit);
             self.emit('data', messageToEmit);
 
-            yield init(cfg);
+            yield exports.init(cfg);
         });
     }, 10000);
 
     if (startRow <= rowCount) {
         let row = msg.body;
-        if( cfg.reader.columns) {
+        if (cfg.reader.columns) {
             const columns = Object.keys(_.keyBy(cfg.reader.columns, 'property'));
             row = _.pick(msg.body, columns);
         }
         stringifier.write(row);
     }
     rowCount++;
     this.emit('end');
```

Comment:
I'm afraid, that `end` is emitted immediately, but actual write will be performed in 10 sec.
What if `admiral` decide to kill the container and actual write won't never happen?

Meanwhile `read.js` https://github.com/elasticio/csv-component/blob/master/lib/read.js#L46 emits `end` after kind of feedback from parser. Can similar logic be implemented here? e.g. listen some event from `stringifier` in order to know when exactly to emit `end`?

Reply:
Unfortunately, ``end`` needs to be emitted here so that this component can get the next message to add to the CSV file.  Moving it to after the the message is emitted won't work.